[ci-app] Document ITR limitations (#15692)
DataDog/documentation: content/en/continuous_integration/intelligent_test_runner/_index.md

By only running tests on relevant code, when tests fail, it is likely a legitimate failure that pertains to the modified code.

## Limitations during beta phase

During the beta phase of Intelligent Test Runner there are certain limitations:

> - Some of the environment variables required in the following sections won't be necessary after beta.

Reviewer: ```suggestion
- Some of the environment variables required in the following sections are only required during the beta phase.
```

> - Intelligent Test Runner only works without the Datadog Agent.
> - There are known limitations on the current implementation of Intelligent Test Runner that would cause tests to be skipped when they shouldn't. If you encounter these scenarios you can add `ITR:NoSkip` (case insensitive) anywhere in your Git commit message to ensure all tests are run. Intelligent Test Runner is not able to detect:
>   - Changes in library dependencies.
>   - Changes in compiler options.
>   - Changes in external services.
>   - Changes to data files in data-driven tests.

Reviewer: ```suggestion
- Intelligent Test Runner only works without the Datadog Agent.
- There are known limitations in the current implementation of Intelligent Test Runner that can cause it to skip tests that should be run.
- Intelligent Test Runner is not able to detect:
  - Changes in library dependencies.
  - Changes in compiler options.
  - Changes in external services.
  - Changes to data files in data-driven tests.
  
To override Intelligent Test Runner and run all tests, add `ITR:NoSkip` (case insensitive) anywhere in your Git commit message. 
```
The override instructions should be listed separately from the ITR limitations.

By only running tests on relevant code, when tests fail, it is likely a legitimate failure that pertains to the modified code.

> ## Limitations during beta phase

Reviewer 2: Can we remove the word "phase" after beta throughout this section?

During the beta of Intelligent Test Runner there are certain limitations:

- Some of the environment variables required in the following sections are only required during the beta.

> - Intelligent Test Runner only works without the Datadog Agent.

Reviewer 3: Can you add some more context to this bullet point? Are there steps they need to take if they were using Datadog Agent and want to implement ITR that they'd need to remove or disable it?
```suggestion
- Intelligent Test Runner only works without the Datadog Agent.
```

Author: I have added some more text

In addition to setting the environment variables above, you need to activate the Intelligent Test Runner on the [Test Service Settings][5] page.

## Setup CI Job

> Intelligent Test Runner uses git metadata information (commit history) to work. However, some CI providers use a git shallow clone (`git clone --depth=0`) which only download the target commit without downloading any historical commit information. This setup does not contain enough information for Intelligent Test Runner. If your CI is using shallow clones, it must be changed.

Reviewer 2: "This setup does not contain enough information for Intelligent Test Runner to work."

> Intelligent Test Runner uses git metadata information (commit history) to work. However, some CI providers use a git shallow clone (`git clone --depth=0`) which only download the target commit without downloading any historical commit information. This setup does not contain enough information for Intelligent Test Runner to work. If your CI is using shallow clones, it must be changed.

Reviewer 3: ```suggestion
Intelligent Test Runner uses git metadata information (commit history) to work. However, some CI providers use a git shallow clone (`git clone --depth=0`) which only downloads the target commit without downloading any historical commit information. This setup does not contain enough information for Intelligent Test Runner to work. If your CI is using shallow clones, it must be changed.
```
